datalad/distributed/create_sibling_ghlike.py

```diff
@@ -0,0 +1,549 @@
+#   -H "Content-Type: application/json" \
+#   -H "Authorization: token <token>" \
+
+# create user repo
+#   -d '{"name": "myrepo", "private": true, "description": "what a dummy?!"}' \
+#   BASE/user/repos
+# properties supported for GH: https://docs.github.com/en/rest/reference/repos#create-a-repository-for-the-authenticated-user
+# overlap with GOGS properties is
+# - name, description, private, auto_init
+# - given auto_init should be False always, this should be enough
+
+# create org repo
+# - GH:   BASE/orgs/<org>/repos
+# - GOGS: BASE/org/<org>/repos
+
+# congruent parts of an API response to a repo creation
+# Example response GitHub
+# {
+#   "name": "createpriv1",
+#   "full_name": "datalad/createpriv1",
+#   "private": true,
+#   "owner": {
+#     "login": "datalad",
+#   },
+#   "description": "what a dummy2?!",
+#   "fork": false,
+#   "created_at": "2021-08-30T07:42:05Z",
+#   "updated_at": "2021-08-30T07:42:05Z",
+#   "html_url": "https://github.com/datalad/createpriv1",
+#   "ssh_url": "git@github.com:datalad/createpriv1.git",
+#   "clone_url": "https://github.com/datalad/createpriv1.git",
+#   "size": 0,
+#   "default_branch": "master",
+#   "permissions": {
+#     "admin": true,
+#     "push": true,
+#     "pull": true
+#   },
+# }
+
+#  Example response GOGS
+# {
+#   "name": "createpriv2",
+#   "full_name": "dataladtesterorg/createpriv2",
+#   "private": false,
+#   "owner": {
+#     "login": "dataladtesterorg",
+#   },
+#   "description": "",
+#   "fork": false,
+#   "created_at": "0001-01-01T00:00:00Z",
+#   "updated_at": "0001-01-01T00:00:00Z",
+#   "html_url": "https://try.gogs.io/dataladtesterorg/createpriv2",
+#   "ssh_url": "git@try.gogs.io:dataladtesterorg/createpriv2.git",
+#   "clone_url": "https://try.gogs.io/dataladtesterorg/createpriv2.git",
+#   "size": 0,
+#   "default_branch": "",
+#   "permissions": {
+#     "admin": true,
+#     "push": true,
+#     "pull": true
+#   }
+# }
+
+import logging
+import re
+from urllib.parse import (
+    urljoin,
+    urlparse,
+)
+import requests
+
+from datalad.downloaders.credentials import Token
+from datalad.downloaders.http import DEFAULT_USER_AGENT
+from datalad.support.param import Parameter
+from datalad.support.constraints import (
+    EnsureChoice,
+    EnsureNone,
+    EnsureStr,
+)
+from datalad.interface.results import get_status_dict
+from datalad.interface.common_opts import (
+    recursion_flag,
+    recursion_limit,
+    publish_depends,
+)
+from datalad.distribution.dataset import (
+    EnsureDataset,
+    require_dataset,
+)
+
+lgr = logging.getLogger('datalad.distributed.create_sibling_ghlike')
+
+
+class _GitHubLike(object):
+    name = None
+    fullname = None
+    create_org_repo_endpoint = None
+    create_user_repo_endpoint = None
+    get_authenticated_user_endpoint = None
+    get_repo_info_endpoint = None
+    response_code_repo_created = requests.codes.created
+    response_code_unauthorized = requests.codes.forbidden
```

Comment:
```suggestion
    # "parametrization" derived classes should provide
    name: str = None
    fullname: str = None
    create_org_repo_endpoint: str = None
    create_user_repo_endpoint: str = None
    get_authenticated_user_endpoint: str = None
    get_repo_info_endpoint: str = None
    response_code_repo_created: int = requests.codes.created
    response_code_unauthorized: int = requests.codes.forbidden
```
type annotation (comes for free since `str` and `int` is built-in) helps to immediately get an idea on what are those to be values to be.  First reading this I was not sure if may be you tried to define `@absractmethods` this ad-hoc way.
Strictly speaking, `""` not `None` should be used or `Optional[str]` with import of Optional from typing should be used, but since we are not using mypy yet to do types checking, I think it would be fine and provide desired DX effect without that.

Reply:
Thx